Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServerTest.cs

```diff
@@ -44,6 +44,15 @@ public class SonarQubeWebServerTest
     private const string ProjectKey = "project-key";
     private const string ProjectBranch = "project-branch";
 
+    [TestMethod]
+    public void Ctor_LogsServerTypeAndVersion()
+    {
+        var logger = new TestLogger();
+        CreateServer(logger: logger);
+
+        logger.AssertInfoMessageExists("Using SonarQube v9.9.");
+    }
```

Comment:
I would add one more test here:

```csharp
    [TestMethod]
    public void Ctor_LogsServerTypeAndVersion()
    {
        var logger = new TestLogger();
        CreateServer(version: new Version("5.42"), logger: logger);

        logger.AssertInfoMessageExists("Using SonarQube v5.42.");
    }
```

Reply:
I'm not sure what's the scenario we want to test with this? To me it looks like testing the test factory method.

Reply:
You are right, I misunderstood how the test is executed. All good!